core/src/main/java/org/mule/runtime/core/privileged/exception/TemplateOnErrorHandler.java

```diff
@@ -315,4 +318,22 @@ public void setRootContainerName(String rootContainerName) {
     updateRootContainerName(rootContainerName, this);
   }
 
+  protected void setFlowLocation(Location location) {
+    this.flowLocation = ofNullable(location);
+  }
+
+  /**
+   * Creates a copy of this ErrorHandler, with the defined location. This location allows to retrieve the
+   * {@link ProcessingStrategy}, and define is a running {@link org.mule.runtime.core.api.transaction.Transaction} is
+   * owned by the {@link org.mule.runtime.core.api.construct.Flow} or {@link org.mule.runtime.core.internal.processor.TryScope}
+   * executing this ErrorHandler.
+   * This is intended to be used when having references to Global ErrorHandlers, since each instance reference
+   * should run with the processing strategy defined by the flow referencing it, and be able to rollback transactions.
+   * @param buildFor
+   * @return copy of this ErrorHandler with location to retrieve {@link ProcessingStrategy}
+   *
+   * @since 4.3.0
```

Comment:
Don't you intend on cherry-picking this?

Reply:
Yes, but since there's no way to know just from seeing the version that a fix included in 4.1.6 is included in 4.2.1 (for example), I intended to cherry pick this modifying the since version (4.2.1, 4.1.6)